Paper background (#4763)
canonical/vanilla-framework: templates/docs/base/paper.md

---
wrapper_template: '_layouts/docs.html'
context:
  title: Paper background | Base elements
---

> # Paper background
> 
> <hr>
> 
> In Vanilla 4.0 we are introducing new option for the background colour of brochure site pages that we call a paper background.
> 
> It replaces the default white page background with light grey `#f3f3f3` colour.
> 
> To use it on the page, add `is-paper` class name to the `body` element.

Reviewer: ```suggestion
# Paper background

<hr>

In Vanilla 4.0 we are introducing a new option for the background colour of brochure site pages that we call a paper background.

It replaces the default white page background with a light grey `#f3f3f3` colour.

Add the `is-paper` class name to the `body` element to use it on the page.
```

```html
<body class="is-paper">
  ...
</body>
```

[See an example page with paper background](/docs/examples/layouts/brochure-site/index).

## Components on paper background

Some components by default assume white page background and don't look well on top of new paper background colour. For those we provide a `on-paper` class name that adjusts the theme of the component to match the new background colour.

The `on-paper` class name can be used on following components:

### Inputs

When you are using text inputs on paper background add `on-paper` class name to the input element.

<div class="embedded-example"><a href="/docs/examples/base/paper/input-on-paper/" class="js-example">
View example of a base form
</a></div>

### Search box

When you are using search box component on top of paper background add `on-paper` class name to `.p-search-box` element.

<div class="embedded-example"><a href="/docs/examples/base/paper/search-box-on-paper/" class="js-example">
View example of a base form

> </a></div>

Author: @ClementChaumel What is this suggestion about? Seems like GH messed it up and it looks like it's replacing whole file…

> [See an example page with paper background](/docs/examples/layouts/brochure-site/index).
> 
> ## Components on paper background
> 
> Some components by default assume white page background and don't look well on top of new paper background colour. For those we provide a `on-paper` class name that adjusts the theme of the component to match the new background colour.
> 
> The `on-paper` class name can be used on following components:
> 
> ### Inputs
> 
> When you are using text inputs on paper background add `on-paper` class name to the input element.
> 
> <div class="embedded-example"><a href="/docs/examples/base/paper/input-on-paper/" class="js-example">
> View example of a base form
> </a></div>
> 
> ### Search box
> 
> When you are using search box component on top of paper background add `on-paper` class name to `.p-search-box` element.
> 
> <div class="embedded-example"><a href="/docs/examples/base/paper/search-box-on-paper/" class="js-example">
> View example of a base form
> </a></div>

Reviewer: ```suggestion
[See an example page with paper background](/docs/examples/layouts/brochure-site/index).

## Components on paper background

Some components by default assume white page background and don't look well on top of the new paper background colour. For those, we provide an `on-paper` class name that adjusts the component's theme to match the new background colour.

The `on-paper` class name can be used on the following components:

### Inputs

When using text inputs on a paper background add the `on-paper` class name to the input element.

<div class="embedded-example"><a href="/docs/examples/base/paper/input-on-paper/" class="js-example">
View an example of a base form
</a></div>

### Search box

When using the search box component on top of the paper background add the `on-paper` class name to the `.p-search-box` element.

<div class="embedded-example"><a href="/docs/examples/base/paper/search-box-on-paper/" class="js-example">
View an example of a base form
</a></div>
```
